README.md

```diff
@@ -23,46 +23,51 @@ python3 -m pip install --extra-index-url https://artifacts.luxonis.com/artifacto
 ### Dependencies
  - cmake >= 3.4
  - C++14 compiler (clang, gcc, msvc, ...)
- - Python
+ - Python3
 
 Along these, dependencies of depthai-core are also required
 See: [depthai-core dependencies](https://github.com/luxonis/depthai-core#dependencies)
 
 
 ### Building
 
-To build a shared library from source perform the following:
+To build and install using pip:
 ```
-git submodule update --init --recursive
-mkdir build && cd build
-cmake .. [-D PYTHON_EXECUTABLE=/full/path/to/python]
-cmake --build . --parallel
+python3 -m pip install .
 ```
-Where `-D PYTHON_EXECUTABLE` option can optionally specify an exact Python executable to use for building.
-
-ℹ️ For the `--parallel` argument, specify a value `[num CPU cores]` or less, to reduce memory consumption during build. E.g.: `--parallel 8`
 
 To build a wheel, execute the following
 ```
 python3 -m pip wheel . -w wheelhouse
 ```
 
-To build and install using pip:
+To build a shared library from source perform the following:
+
+> ℹ️ To speed up build times, use `cmake --build build --parallel [num CPU cores]` (CMake >= 3.12).
+For older versions use: Linux/macOS: `cmake --build build -- -j[num CPU cores]`, MSVC: `cmake --build build -- /MP[num CPU cores]`
+
 ```
-python3 -m pip install .
+git submodule update --init --recursive
+cmake -H. -Bbuild
+cmake --build build
 ```
+To specify custom Python executable to build for, use `cmake -H. -Bbuild -D PYTHON_EXECUTABLE=/full/path/to/python`.
+
 
 ## Running tests
 
 To run the tests build the library with the following options
 ```
 git submodule update --init --recursive
-mkdir build_tests && cd build_tests
-cmake .. -D DEPTHAI_PYTHON_ENABLE_TESTS=ON -D DEPTHAI_PYTHON_ENABLE_EXAMPLES=ON -D DEPTHAI_PYTHON_TEST_EXAMPLES=ON
-cmake --build . --parallel
-ctest
+cmake -H. -Bbuild -D DEPTHAI_PYTHON_ENABLE_TESTS=ON -D DEPTHAI_PYTHON_ENABLE_EXAMPLES=ON -D DEPTHAI_PYTHON_TEST_EXAMPLES=ON
+cmake --build build
 ```
 
+Then navigate to `build` folder and run `ctest`
+```
+cd build
```

Comment:
Mention individual example testing, with custom timeout:
`TEST_TIMEOUT=0 ctest -R "01_rgb_preview" --verbose`

Reply:
Good idea, will add this back here